src/main/java/teammates/ui/controller/InstructorCourseDetailsPageData.java

```diff
@@ -20,9 +24,132 @@ public InstructorCourseDetailsPageData(AccountAttributes account) {
+        this.courseDetails = courseDetailsParam;
+        this.instructors = instructorsParam;
+        
+        boolean isDisabled = !this.currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS);
+        this.giveCommentButton = createButton("<span class=\"glyphicon glyphicon-comment glyphicon-primary\"></span>",
+                                              "btn btn-default btn-xs icon-button pull-right", "button_add_comment", 
+                                              null, "", "tooltip", null, isDisabled);
+  
+        isDisabled = !this.currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+        String onClick = "if(toggleSendRegistrationKeysConfirmation('" + Sanitizer.sanitizeForJs(this.courseDetails.course.id) + "')) "
+                                + "window.location.href='" + Sanitizer.sanitizeForJs(this.getInstructorCourseRemindLink()) + "';";
+        
+        this.courseRemindButton = createButton(null, "btn btn-primary", "button_remind", null, 
+                                               Const.Tooltips.COURSE_REMIND, "tooltip", onClick, isDisabled);
+        
+        this.studentsTable = new CourseDetailsStudentsTable();
+        int idx = 0;
+        for (StudentAttributes student : students) {
+            CourseDetailsStudentsTableRow row = new CourseDetailsStudentsTableRow();
+            row.student = student;
+            
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(student.section, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS);
+            ElementTag viewButton = createButton("View", "btn btn-default btn-xs", null, this.getCourseStudentDetailsLink(student),
+                                                 Const.Tooltips.COURSE_STUDENT_DETAILS, "tooltip", null, isDisabled);
+            
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+            ElementTag editButton = createButton("Edit", "btn btn-default btn-xs", null, this.getCourseStudentEditLink(student),
+                                                 Const.Tooltips.COURSE_STUDENT_EDIT, "tooltip", null, isDisabled);
+            
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+            ElementTag sendInviteButton = createButton("Send Invite", "btn btn-default btn-xs", null, this.getCourseStudentRemindLink(student),
+                                                 Const.Tooltips.COURSE_STUDENT_REMIND, "tooltip", "return toggleSendRegistrationKey()", isDisabled);
+       
+            isDisabled = !this.currentInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+            onClick = "return toggleDeleteStudentConfirmation('" + sanitizeForJs(student.course) + "','" + sanitizeForJs(student.name) + "')";
```

Comment:
Most of these are easily exceeding the characters per line limit.
Split them into more, we want production level code even if this is a dev branch.